code/modules/events/anomaly.dm

```diff
@@ -1,23 +1,32 @@
+#define TURF_FIND_TRIES 10
+
 /datum/event/anomaly
 	name = "Anomaly: Energetic Flux"
 	var/obj/effect/anomaly/anomaly_path = /obj/effect/anomaly/flux
+	var/turf/target_turf
 	announceWhen = 1
 
 /datum/event/anomaly/setup()
-	impact_area = findEventArea()
-	if(!impact_area)
-		CRASH("No valid areas for anomaly found.")
-	var/list/turf_test = get_area_turfs(impact_area)
-	if(!length(turf_test))
+	for(var/tries in 0 to TURF_FIND_TRIES)
+		impact_area = findEventArea()
+		if(!impact_area)
+			CRASH("No valid areas for anomaly found.")
+		var/list/candidate_turfs = get_area_turfs(impact_area)
+		while(length(candidate_turfs))
+			var/turf/candidate = pick_n_take(candidate_turfs)
+			if(!is_blocked_turf(candidate,TRUE))
+				target_turf = candidate
+				break
+		if(target_turf)
+			break
+	if(!target_turf)
 		CRASH("Anomaly: No valid turfs found for [impact_area] - [impact_area.type]")
```

Comment:
this will only list the last area it tries. might be worth while to put it in the for loop. I doubt this will ever fire however unless a player fills a room with walls or something stupid.

Reply:
But if I do it will invalidate the whole point of the loop by exiting it early. 
I could use stack_trace() but it'll just end noise to the runtime logs for something that is not really an error (since we are trying to do it 10 times).

In my opinion it should only get logged if we really tried all 10 times and failed.

Also failure to find the turf on the first try is a lot more likely, since I'm not just checking for walls but for dense objects in general.

I'll write an error message that better reflects that the failure is final though.